app/controllers/games_controller.rb

```diff
@@ -19,7 +19,48 @@ def show
 
   def update
     @game = Game.find(params[:id])
-    @game.update_attributes(game_params)
+
+    if request.xhr?
+      # e.g. pawnBlack or kingWhite, etc
+      selected_piece = request['selected_piece']
+
+      # Square that the selected_piece initially
+      # resides in. Format is row4col4, or row1col6, etc
+      initial_square = request['initial_square']
+
+      # Square that the user wants the selected piece
+      # to move to. Format is row4col4, or row1col6, etc
+      final_square = request['final_square']
+
+      # Create backend logic in models and call them here
+      # with the incoming data as arguments
+
+      # data to be returned. Will contain the return of
+      # the model logic.
+      # Piece_captured is so javascript can update the
+      # captured pieces board.
+      # Message would go in the game status board, so
+      # stuff like 'your turn' or 'invalid move',
+      # 'checkmate', etc.
+      valid_move = true
+      piece_captured = 'pawnWhite'
+      message = nil
+
+      # feel free to add new data entries to the output
+      respond_to do |format|
+        format.json do
+          render json: {
+            validMove: valid_move,
+            pieceCaptured: piece_captured,
+            message: nil
+          }.to_json
+        end
+      end
+
+    else
+      # This code runs if the request is not ajax
+      @game.update_attributes(game_params)
+    end
```

Comment:
hmmm. I sort of follow you but sort of don't. Since we don't have a pieces controller, should I set up a controller for that and then hook up a route for it in routes.rb? I also don't have the front end pieces hooked up to the backend at all, so I think i'd still need to send the original position as well so the backend can find and match what piece/position the ajax request is referring to.